scripts/model_upload.py

```diff
@@ -31,14 +33,79 @@ def upload_part(url, params, headers, data, parts, verify):
     parts.append({'ETag': part_etag, 'PartNumber': params['partNumber']})
 
 
+def upload(url, params, headers, part_size, verify=False):
+    file_path = params['file_path']
+    if os.path.isfile(file_path):
+        if tarfile.is_tarfile(file_path):
+            untar_and_upload(url, params, headers, part_size, verify)
+        else:
+            upload_model(url, params, headers, part_size, verify)
+    elif os.path.isdir(file_path):
+        upload_dir(url, params, headers, part_size, verify)
+    else:
+        raise Exception("Unrecognized type of upload")
+
+
+def untar_and_upload(url, params, headers, part_size, verify=False):
+    tar = tarfile.open(params['file_path'])
```

Comment:
what will happen if there will be plenty of files in tar, instead of one directory?

Reply:
Like I said lower, I assume a structure as in TFS